### PR TITLE
Feat/vendor sidebar

### PR DESCRIPTION
**Pull request by MargaritaSukhova**
## Describe your changes

- Refactored vendor side bar component to use navLink and isActive prop to update styles when menu button is selected
- Fixed mocks 

## Issue ticket code (and/or) and link

- [Link to JIRA ticket](https://2024internship.atlassian.net/browse/SCRUM-93)

### **General**

- [x] Assigned myself to the PR
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [x] Updated the documentation
- [x] Performed a self-review of my code
- [x] Types for input and output parameters
- [x] Don't have "any" on my code
- [x] Used the try/catch pattern for error handling
- [x] Don't have magic numbers
- [x] Compare only with constants not with strings
- [x] No ternary operator inside the ternary operator
- [x] Don't have commented code
- [x] no links in the code, env links should be in env file (for example: server url), constant links (for example default avatar URL) should be in constant file.
- [x] Used camelCase for variables and functions
- [x] Date and time formats are on the constants
- [x] Functions are public only if it's used outside the class
- [x] No hardcoded values
- [ ] covered by tests
- [x] Check your commit messages meet the [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/).

### Frontend

- [x] Components and business logic are separated
- [x] Colors, Font Size, and Font Name is on the theme or in the constants
- [x] No text in the components, use i18n approach
- [x] No inline styles
- [x] Imports are absolute
- [x] Attach a screenshot if PR has visual changes.
Снимок экрана 2024-07-30 в 22 03 44
Снимок экрана 2024-07-30 в 22 03 51
Снимок экрана 2024-07-30 в 22 04 00
Снимок экрана 2024-07-30 в 22 04 08
Снимок экрана 2024-07-30 в 22 04 17
Снимок экрана 2024-08-06 в 23 53 26

### CODE REVIEW

File: src/components/shared/ImageForm/index.tsx [hunk 1]
```diff
@@ -28,18 +28,20 @@ function ImagesForm() {
...
-              <ImageCard
-                type={item.type}
-                src={item.src}
-                isPrimary={item.isPrimary}
-              />
-            )
-        )}
-        {images.filter((item) => item.type === 'image').length <
-          maxNumberImage && <ImageCard type="image" />}
+        {images &&
```
Comment (lyudad) [inline src/components/shared/ImageForm/index.tsx]: images?.map(

File: src/components/shared/ImageForm/index.tsx [hunk 2]
```diff
@@ -28,18 +28,20 @@ function ImagesForm() {
...
+            (item) =>
+              (item.type === 'image' || item.type === 'video') && (
+                <ImageCard
+                  type={item.type}
+                  src={item.src}
+                  isPrimary={item.isPrimary}
+                />
+              )
+          )}
+        {images &&
```
Comment (lyudad) [inline src/components/shared/ImageForm/index.tsx]: images?.filter

Comment (MargaritaSukhova) [reply src/components/shared/ImageForm/index.tsx]: fixed

File: src/components/shared/ImageForm/index.tsx (hunk 1 above)
Comment (MargaritaSukhova) [reply src/components/shared/ImageForm/index.tsx]: fixed

File: src/pages/vendor/VendorLayout/VendorSideBar/index.tsx [hunk 3]
```diff
@@ -63,229 +60,255 @@ function VendorSideBar() {
...
+          <NavLink to={urls.VENDOR_CHATS}>
+            {({ isActive }) => (
+              <StyledListItemButton selected={isActive}>
+                <Box
+                  display="flex"
+                  justifyContent="flex-start"
+                  alignItems="center"
+                  width="140px"
+                >
+                  {isActive ? (
```
Comment (lyudad) [inline src/pages/vendor/VendorLayout/VendorSideBar/index.tsx:223]: find the way to do it with binary operation, its over code to add empty <> each tyme

Comment (MargaritaSukhova) [reply src/pages/vendor/VendorLayout/VendorSideBar/index.tsx:223]: fixed unreadChatMessages

**Review by lyudad — APPROVED** [review]

**Pull request merged**